python/modules/connection.py

```diff
@@ -4,220 +4,222 @@
+        async with session.post(receiver_endpoint, data=serialized_msg) as resp:
+            print(resp.status)
+            print(await resp.text())
 
-    # Get their did and vk and store in wallet
-    their_did = resp_data["did"]
-    their_vk = resp_data["verkey"]
+    return Message(
+        type=UI_NEW.INVITE_SENT,
+        id=agent.ui_token,
+        content={'name': conn_name})
 
-    identity_json = json.dumps({
-        "did": their_did,
-        "verkey": their_vk
-    })
 
-    await did.store_their_did(wallet_handle, identity_json)
-    #TODO: Do we want to store the metadata of owner and endpoint with their did?
+async def invite_received(msg: Message, agent) -> Message:
+    conn_name = msg.content['name']
 
-    # Create pairwise identifier
-    await pairwise.create_pairwise(
-        wallet_handle,
-        their_did,
-        my_did,
-        json.dumps({"test": "this is metadata"})
+    return Message(
+        type=UI_NEW.INVITE_RECEIVED,
+        content={'name': conn_name,
+                 'endpoint': msg.content['endpoint'],
+                 'history': msg}
     )
 
 
+async def send_request(msg: Message, agent) -> Message:
+    their_endpoint = msg.content['endpoint']
+    conn_name = msg.content['name']
+    connection_key = msg.content['key']
 
+    my_endpoint_uri = agent.endpoint
 
-async def handle_request_accepted(request):
-    """ Handle reception of accept connection request message.
-    """
-    accept_did = request.match_info['did']
-    agent = request.app['agent']
-    wallet_handle = agent.wallet_handle
-
-    if accept_did not in agent.received_requests:
-        raise web.HTTPNotFound()
-
-    msg = Serializer.unpack(agent.received_requests[accept_did])
+    (endpoint_did_str, endpoint_key) = await did.create_and_store_my_did(agent.wallet_handle, "{}")
 
-    #TODO: validate correct format for incoming data
-    data = msg.data
-    endpoint = data['endpoint']
-    verkey = data['verkey']
-    owner = data['owner']
-
-    ident_json = json.dumps(
+    #  workaround to pass conn_name securely back to the invite sender, otherwise we could just send the did
+    data_to_send = json.dumps(
         {
-            "did": accept_did,
-            "verkey": verkey
+            "did": endpoint_did_str,
+            "conn_name": conn_name
         }
     )
 
+    endpoint_did_bytes = str_to_bytes(data_to_send)
+
     meta_json = json.dumps(
         {
-            "owner": owner,
-            "endpoint": endpoint
+            "conn_name": conn_name
         }
     )
 
-    (my_did, _) = await did.create_and_store_my_did(wallet_handle, "{}")
-
-    await did.store_their_did(wallet_handle, ident_json)
+    await did.set_did_metadata(agent.wallet_handle, endpoint_did_str, meta_json)
 
-    await did.set_endpoint_for_did(wallet_handle, accept_did, endpoint, verkey)
+    inner_msg = Message(
+        type= "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/connections/1.0/request",
```

Comment:
For consistency, these types should be referenced in the same way the other type names have been.